scrapy_zyte_api/providers.py

```diff
@@ -150,6 +154,10 @@ async def __call__(  # noqa: C901
             cls_stripped = strip_annotated(cls)
             assert isinstance(cls_stripped, type)
             kw = item_keywords.get(cls_stripped)
+            if cls_stripped is Geolocation and is_typing_annotated(cls):
+                item = AnnotatedResult(Geolocation(), cls.__metadata__)  # type: ignore[attr-defined]
+                results.append(item)
```

Comment:
Is there a need to update cache?

Reply:
As I understand it the purpose of the cache is to skip requesting things, and we don't request Geolocation, but also we don't want to remove it from to_provide like it would be done if it's found in the cache. On the other hand  I don't know in which cases is the cache currently used.

Reply:
Also is it possible to request a Product without a Geolocation, cache it and then get that instance when the provider code is called again with a Geolocation, thus getting a wrong one?